### ISSUE TITLE
RStudio desktop segfaults on Ubuntu 18.04

### ISSUE THREAD

**Issue by fsaintjacques** [rstudio/rstudio#4271]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.1.463
    OS Version      : Ubuntu 18.04.1 LTS
    R Version       :  version 3.4.4 (2018-03-15)

### Steps to reproduce the problem

Undock the laptop from docking station.

### Describe the problem in detail

rstudio process segfaults (not always).

```
[142214.335961] rstudio[2216]: segfault at 28 ip 00007f498239d711 sp 00007ffefd6097f0 error 4 in libQt5Widgets.so.5.4.2[7f49821cd000+686000]
```
No stack trace is provided by apport.

**Comment by HFDave** [comment]
I have been running the same setup on several laptops and desktops for more than half a year without any problems. I have never ever witnessed RStudio to segfault and it also wakes up from sleep with no problems at all. The problem can be hardware specific.

**Comment by kevinushey** [comment]
Have you tried the [preview release](https://www.rstudio.com/products/rstudio/download/preview/)?

**Comment by nicku33** [comment]
I have this as well.

> Feb 13 13:57:07 nu-t440s kernel: [315509.103212] rstudio[30153]: segfault at 28 ip     
00007fab97768711 sp 00007ffe59e3bed0 error 4 in libQt5Widgets.so.5.4.2[7fab97598000+   
686000]

I system it's a problem with our version of libQt5. I *think* that's a shared library and can move at it's own version number. Google "error 4 in libQt5Widgets.so.5.4.2" to see other people w same issue:

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=790825
https://bugs.launchpad.net/stellarium/+bug/1522905

I will try changing the Qt version.

**Comment by nicku33** [comment]
Ah they are not using system installed Qt but come with it's own:
> ls /usr/lib/rstudio/libQt*
> ..
> libQt5Widgets.so.5.4.2

**Comment by nicku33** [comment]
What's strange is this would seem to imply we should be seeing 5.11 with latest releases:
> https://github.com/rstudio/rstudio/blob/master/dependencies/linux/install-qt-sdk

**Comment by fsaintjacques** [comment]
great find @nicku33 I can fully reproduce by undocking my laptop, rstudio immediately segfaults.

**Comment by HFDave** [comment]
Do you use Nvidia card and drivers?

**Comment by fsaintjacques** [comment]
This is fixed in the preview release.

**Comment by fsaintjacques** [comment]
@HFDave no, this laptop only has intel integrated graphics.

**Comment by fsaintjacques** [comment]
It turns out that sleep/lid was just a symptom of me removing the dock it seems.

**Comment by nicku33** [comment]
@fsaintjacques Should the predone builds available from rstudio.com have qt5.4.2 bundled ? That seems rather old, but it's what is on the site for debian...

"""
wget https://download1.rstudio.org/rstudio-1.1.463-amd64-debian.tar.gz
tar -xf rstudio-1.1.463-amd64-debian.tar.gz
ls rstudio-1.1.463/bin/*5.4*

rstudio-1.1.463/bin/libQt5Core.so.5.4.0
...
"""

**Comment by fsaintjacques** [comment]
```
iota:Downloads/ $ dpkg -c rstudio-xenial-1.1.463-amd64.deb |grep libQt5Core
-rwxr-xr-x root/root   5522936 2017-05-02 19:02 ./usr/lib/rstudio/bin/libQt5Core.so.5.4.2
lrwxrwxrwx root/root         0 2018-10-24 20:19 ./usr/lib/rstudio/bin/libQt5Core.so.5 -> libQt5Core.so.5.4.2
iota:Downloads/ $ dpkg -c rstudio-1.2.1268-amd64.deb  | grep libQt5Core
-rwxr-xr-x root/root   7931408 2019-02-04 14:58 ./usr/lib/rstudio/lib/libQt5Core.so.5
```

**Comment by nicku33** [comment]
Oops you're right my bad. To summarize:
 * qt 5.4.2 (and possibly earlier) has this issue with segfaulting on wake from sleep, at least for some people, as can be seen by bug reports from others
 * the current version on rstudio for debian on the rstudio site has qt 5.4.2 in the .dev and 5.0 in the tarball versions
 * Rstudio 1.2, currently in preview no longer uses symlinks to minor dot versions but has an updated qt, presumably > 5.4.2 which should resolve this issue (I'm trying to resolve come clang dependencies to confirm this)

**Comment by kevinushey** [comment]
RStudio 1.1 indeed uses the (very old) Qt 5.4.2, which has these sorts of known crashes. In RStudio v1.2, we plan to upgrade to Qt 5.12.1 (or on older OSes where that's not supported, Qt 5.10.1) which should resolve this particular issue.

**Comment by FabianUx** [comment]
Thank you for the research.
I was experiencing the same issue when suspending the computer:
Linux Mint 19.1 (NVidia and Intel graphic cards)
Version 1.1.463 – RStudio
Mozilla/5.0 (X11; N; Linux x86_64) AppleWebKit/538.1 (KHTML, like Gecko) RStudio Safari/538.1 Qt/5.4.2

**Comment by kevinushey** [comment]
Please try the v1.2 [preview release](https://www.rstudio.com/products/rstudio/download/preview/).

**Comment by FabianUx** [comment]
Thanks
I'm trying to do so (I just need to make sure I can test version 1.2 while still working with the previous release).
Any risk of conflicts?
Are they installed in different folders?

**Comment by FabianUx** [comment]
It works in Version 1.2.1303. Rstudio remains open after suspension whether there was a widget active or not.

**Comment by jmcphers** [comment]
Sounds like this is resolved in 1.2 with our new version of Qt; please reopen if you're still having trouble with the new release!

**Comment by ferroao** [comment]
Answer in https://askubuntu.com/questions/1041588/virtualbox-not-launching-on-ubuntu-18-04-qt-lib-problem solved for me
